plutus-core/executables/debugger/Event.hs

```diff
@@ -84,23 +90,39 @@ handleDebuggerEvent _ (B.AppEvent (UpdateClientEvent budgetData cekState)) = do
                   -- is the line break, hence the `- 1`.
                   _hcELoc = Just $ B.Location (srcSpanELine uplcSpan, srcSpanECol uplcSpan - 1)
                 }
-    let sourceHighlight :: Maybe HighlightSpan = do
+
+
+    let msourceSrcSpan :: Maybe SrcSpan = do
             txSpans <- txAnn <$> cekStateAnn cekState
             -- FIXME: use some/all spans for highlighting, not just the first one
+            -- because now we arbitrary selected the first (in-order) srcsspan
             firstTxSpan <- S.lookupMin $ coerce txSpans
-            -- TODO: the HS_FILE supplied from the command line gets highlighted
-            -- The highlighting will not make sense or even break if the user provides
-            -- wrong HS_FILE or if the UPLC originated from multiple HS modules.
+            pure firstTxSpan
+
+    let msourceHighlight :: Maybe HighlightSpan = do
+            sourceSrcSpan <- msourceSrcSpan
             pure HighlightSpan
-                { _hcSLoc = B.Location (srcSpanSLine firstTxSpan, srcSpanSCol firstTxSpan),
+                { _hcSLoc = B.Location ( srcSpanSLine sourceSrcSpan
+                                       , srcSpanSCol sourceSrcSpan
+                                       ),
                   -- GHC's SrcSpan's ending column is one larger than the last character's column.
                   -- See: ghc/compiler/GHC/Types/SrcLoc.hs#L728
-                  _hcELoc = Just $ B.Location (srcSpanELine firstTxSpan, srcSpanECol firstTxSpan -1)
+                  _hcELoc = Just $ B.Location ( srcSpanELine sourceSrcSpan
+                                              , srcSpanECol sourceSrcSpan -1
+                                              )
                 }
+    -- the current sourcespan may have jumped to another hs file, update the text of the editor
+    msourceText <- liftIO $ for (liftA2 (</>) hsDir $ srcSpanFile <$> msourceSrcSpan) Text.readFile
+                           `catchError` \ _ -> pure mempty
```

Comment:
What is so suspicious about it? This is the most simple solution. If you fail to find the file (insuficcient rights or file does not exist) then don't return text. But if you have a better and not verbose approach to do file I/O let me know

Reply:
Swallowing errors is almost never a good idea IME. It's all fine until it's not working and you don't know why because the error didn't get displayed anywhere. Ideally we would show it somewhere or log it.

Reply:
I understand. I think cannot use stdout because i am in "ncurses-style" mode; I could potentially use stderr but I don't want to risk how it will be redirected. So, I will use the same brick editor widget for now to display some error text.

Reply:
Actually it is better to use the logging editortext for the error. I will do that tomorrow.